app/views/contexts/section_summary_context.py

```diff
@@ -118,10 +134,50 @@ def _build_summary(self, return_to: Optional[str]):
                 )
             }
 
-            return {**collapsible, **summary_elements}
+        else:
+            summary_elements = {}
+
+        all_groups = []
+        group_number = 0
+        for group in list(self.section["groups"]):
+
+            group_name = group["id"]
+            standard_group: list = []
+            list_collector_group = []
+            for block in group["blocks"]:
+                if block["type"] == "ListCollector":
+                    if standard_group:
+                        partial = {
+                            "id": f"{group_name}-{group_number}",
+                            "blocks": standard_group,
+                        }
+                        all_groups.append(partial)
+                        group_number += 1
+                    list_collector_group.append(block)
+                    partial = {
+                        "id": f"{group_name}-{group_number}",
+                        "blocks": list_collector_group,
+                    }
+                    all_groups.append(partial)
+                    group_number += 1
+                    list_collector_group = []
+                    standard_group = []
+
+                else:
+                    standard_group.append(block)
+
+            partial = {
+                "id": f"{group_name}-{group_number}",
+                "blocks": standard_group,
+                "title": group.get("title"),
+            }
+
+            all_groups.append(partial)
```

Comment:
I find this block of code quite hard to follow. I think it could be made clearer if it was extracted into it own method that returns the groups. Also some of the vars could be renamed to make them clearer like `standard_group` and `partial`, and maybe a doctring to make it clearer. Some vars type hints are missing too.

Reply:
I moved it to a separate method, renamed the vars so they hopefully make more sense and added a lot of comments, some of them might be unnecessary or too obvious but I wanted to start somewhere. Thanks1